ministryofjustice/laa-review-criminal-legal-aid
# [CRIMAPP-91] Supervisors can update a caseworkers competencies

## Description of change
Adds a form which enables supervisors to click on a caseworker and set/unset their competencies as desired. Also introduces event sourcing the history of changes to a caseworkers competencies, storing the caseworker id, supervisor id (of the supervisor who made the change) and the updated competencies. A follow up PR will display the competency event history. 

## Link to relevant ticket
[CRIMAPP-91](https://dsdmoj.atlassian.net/browse/CRIMAPP-91)

## Notes for reviewer

## Screenshots of changes (if applicable)

### Before changes:

### After changes:
<img width="1310" alt="Screenshot 2023-11-13 at 11 24 49" src="https://github.com/ministryofjustice/laa-review-criminal-legal-aid/assets/51047911/c9bfee40-1739-4f66-a9c1-3afc93b69470">
<img width="1310" alt="Screenshot 2023-11-13 at 11 23 44" src="https://github.com/ministryofjustice/laa-review-criminal-legal-aid/assets/51047911/deec42bb-5b59-4550-aeea-c26fc4acbc00">
<img width="1310" alt="Screenshot 2023-11-13 at 11 23 27" src="https://github.com/ministryofjustice/laa-review-criminal-legal-aid/assets/51047911/2227ac9a-7075-44af-ad70-190ab11ee203">
<img width="1310" alt="Screenshot 2023-11-13 at 11 23 34" src="https://github.com/ministryofjustice/laa-review-criminal-legal-aid/assets/51047911/becc7895-f891-4c85-807d-56be1c5eeef9">

## How to manually test the feature
Requires you to have at least one supervisor and caseworker user.

- Log into review as a supervisor. You should see the manage competencies tab and be able to click onto it to view the dashboard. There you should be able to see a list any active caseworkers with 'No competencies' displayed in the competencies column as they are not yet set.
- Click on a caseworker and you should see the manage competencies form with three competencies. Test that you can set and unset the competencies of the caseworker. You should be redirected to the manage competencies dashboard where you can see the competencies displayed. 
- Check that you can unset all competencies and this should display the placeholder text 'No competencies' again


[CRIMAPP-91]: https://dsdmoj.atlassian.net/browse/CRIMAPP-91?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
